### PR TITLE
Modestas gujis/feature/t 4

### PR DESCRIPTION
**Pull request by ModestasGujis**

### CODE REVIEW

File: cypress/integration/ClientCreation.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+        const randomName = Math.random().toString(36).substring(9);
+
+        ClientsPage.getCreateButton().click()
+
+        ClientsPage.getOrganizationField().type(randomName)
+        ClientsPage.getFirstNameField().type('first name')
+        ClientsPage.getLastNameField().type('Last name')
+        ClientsPage.getEmailField().type('my@email.com')
+        ClientsPage.getSaveButton().click()
```
Comment (DavidV44) [inline cypress/integration/ClientCreation.js]: It doesn't look like that you check that client you just created is actually created

**Review by DavidV44 — CHANGES_REQUESTED** [review]

File: cypress/integration/ClientCreation.js (hunk 1 above)
Comment (ModestasGujis) [reply cypress/integration/ClientCreation.js]: I believe it's fixed.

File: cypress/helpers/TaskPage.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    getCreateButton: () => cy.contains('Create Task'),
+
+    getTaskNameInput: () => cy.get('[id="taskDetailsForm.name"]'),
+    
+    getDescriptionInput: () => cy.get('[id="taskDetailsForm.description"]'),
+
+    getBillToClient: () => cy.get('.Select-value'),
+
+    getBillToClientOption: (option) => cy.get('[aria-label="' + option + '"]'), // 'Yes' or 'No'
```
Comment (nikolaj-tolkaciov) [inline cypress/helpers/TaskPage.js]: unnecessary comment

File: cypress/helpers/TimeLoggingPage.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+    getTileForm: () => cy.get('.tile.form'),
+
+    getActiveTab: () => cy.get('.main-nav__link--active'),
+
+    getUserName: () => cy.get('.user-info__title'),
+
+    getTabs: () => cy.get('.main-nav').find('li'),
+
+    getCurrDay: () => cy.get('.calendar--today > span')
```
Comment (nikolaj-tolkaciov) [inline cypress/helpers/TimeLoggingPage.js]: why you are not using above approach here as well? 
 getTabs: () => cy.get('.main-nav').find('li'),

File: cypress/integration/ClientCreation.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import ClientsPage from '../helpers/ClientsPage'
+
+// T-4
```
Comment (nikolaj-tolkaciov) [inline cypress/integration/ClientCreation.js]: unnecessary comment

File: cypress/integration/ClientCreation.js (hunk 1 above)
Comment (nikolaj-tolkaciov) [reply cypress/integration/ClientCreation.js]: Don't see any assertions for created client

File: cypress/integration/adminTasks.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import TaskPage from '../helpers/TaskPage'
+
+// T-3
```
Comment (nikolaj-tolkaciov) [inline cypress/integration/adminTasks.js]: unnecessay comment

File: cypress/integration/login.js [hunk 6]
```diff
@@ -1,27 +1,43 @@
...
+
+            const activeTab = TimeLoggingPage.getActiveTab()
+            
+            activeTab.should('have.length', 1)
+            activeTab.should('contain', 'Time Logging')
+
+            TimeLoggingPage.getUserName().should('contain', 'Modestas Gujis')
+            TimeLoggingPage.getTabs().should('have.length', tabCount[idx])
+
+            TimeLoggingPage.getActiveTab().should('have.css', 'color', 'rgb(64, 76, 237)')
```
Comment (nikolaj-tolkaciov) [inline cypress/integration/login.js]: you have a variable for this in a case above, why don't you reuse it here too?

File: cypress/integration/login.js [hunk 7]
```diff
@@ -1,27 +1,43 @@
...
-        
-        cy.visit('/');
-        cy.get('.Select.not-valid').should('not.visible')
-        cy.get('[type="submit"]').click();
-        cy.get('.Select.not-valid').should('be.visible')
+import LoginPage from '../helpers/LoginPage'
+import TimeLoggingPage from '../helpers/TimeLoggingPage'
+import { ROLES, tabCount } from '../helpers/constants'
+
+// T-2
```
Comment (nikolaj-tolkaciov) [inline cypress/integration/login.js]: unnecessary comment

File: cypress/integration/adminTasks.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        TaskPage.getHourlyRate().type(Math.random() * 15)
+
+        TaskPage.getSubmitButton().click()
+        
+        cy.url().should('include', '/tasks')
+        cy.url().should('not.include', '/create')
+
+        TaskPage.visit()
+		TaskPage.getNameFilter().type(randomName)
+		TaskPage.getTaskList().should('have.length', 1)
```
Comment (nikolaj-tolkaciov) [inline cypress/integration/adminTasks.js]: indent issues

**Review by nikolaj-tolkaciov — CHANGES_REQUESTED** [review]
Good code! But according to the convection file names should start with lower case, there is few ways to handle the gaps though

**Review by DavidV44 — APPROVED** [review]

**Pull request merged**